src/FINDOLOGIC/Api/ResponseObjects/Response.php

```diff
@@ -13,6 +26,37 @@ abstract class Response
      */
     abstract public function __construct($response, $responseTime = null);
 
+    /**
+     * Creates a new Response instance based on the given request builder.
+     *
+     * @param RequestBuilder $requestBuilder
+     * @param GuzzleResponse $response
+     * @param GuzzleResponse|null $alivetestResponse The alivetest response, or null if no alivetest was made.
+     * @param float|null $responseTime
+     * @return Response
+     */
+    public static function getInstance(
```

Comment:
I think `buildInstance` or just `build` would be more suitable since you create something new, not getting something that already exists.